### PR TITLE
Add World & Chunk implementation

### PR DESCRIPTION
**Pull request by 0ddlyoko**
This PR adds basic stuff for managing Chunk & World

Waiting for:
- ~~#44~~
- #48 
- ~~#49~~

### CODE REVIEW

**Comment by Geolykt** [comment]
Chunks already exist at https://github.com/tazz4843/McPy/blob/develop/classes/BasicClasses.py#L37

File: classes/WorldGenerator.py [hunk 1]
```diff
@@ -347,33 +350,33 @@ def noise3(self, x, y, z):
...
                 blockY = scaleNoise(bY, (63, 80))  # Scale the noise to be between min-max y value
                 positions.append((blockX, blockY, blockZ))
-        chunk = Chunk(x, y, z, [], region, width, height)
-        await self._regenerate_chunk(x, y, z, region, positions, chunk)
+        await self._regenerate_chunk(x, y, z, positions, chunk)
         return chunk
 
-    async def _regenerate_chunk(self, x, y, z, region, positions, chunk: Chunk):
+    async def _regenerate_chunk(self, x, y, z, positions, chunk: Chunk):
+        # WTF ?
```
Comment (Geolykt) [inline classes/WorldGenerator.py]: If I remember it correctly I did that to make the code run faster as you shouldn't confuse ChunkColum (which will be added later) and Chunk (also known as ChunkSection), this regenerates a single 16 x 16 x 16 Chunk section, which results in the fact that positions needs to be regenerated pretty often. Now I reckon that you could simply limit the size of the positions array, but If I recall correct there was another limitation with (tree?) generation that required the full positions array to be known.

Maybe the generators should just be generating everything after the initial generation, I suppose that that would actually be more efficent in terms of inter-chunk structure generation or this one

Comment (ntoskrnl4) [reply classes/WorldGenerator.py]: The vanilla game works by first generating the world's overall terrain and shape (mesa goes here, mountain is at y142, etc). Then it adds basic world features like caves, rivers, oceans, water lakes, etc. and then does more complex non-natural features like villages, strongholds, and the like

Comment (Geolykt) [reply classes/WorldGenerator.py]: That's what I have in mind, which we would need to do later on

Comment (0ddlyoko) [reply classes/WorldGenerator.py]: I've deleted the commit that has changed this class, I'll edit it later

**Comment by 0ddlyoko** [comment]
Waiting for #44 to have network (to be able to send chunks)

**Comment by 0ddlyoko** [comment]
Waiting for #48 (to be able to send chunks)

**Comment by 0ddlyoko** [comment]
Also waiting for #49 (to be able to load chunk asynchronously & to avoid editing main.py file)

**Comment by hydrostaticcog** [comment]
Developer Left Project

**Pull request closed without merging**